cmd/stats-exporter/main.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"database/sql"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/letsencrypt/boulder/cmd"
+)
+
+type sqlRows interface {
+	Next() bool
+	Scan(dest ...interface{}) error
+}
+
+type dbQueryable interface {
+	Query(string, ...interface{}) (*sql.Rows, error)
+}
+
+var sqlOpen = func(driver, dsn string) (dbQueryable, error) {
+	return sql.Open(driver, dsn)
+}
+
+var execRun = func(c *exec.Cmd) error {
+	return c.Run()
+}
+
+func queryDB(dbConnect, beginTimeStamp, endTimeStamp string) (*sql.Rows, error) {
+	dbDSN, err := ioutil.ReadFile(dbConnect)
+	if err != nil {
+		return nil, fmt.Errorf("Could not open database connection file: %s", err)
+	}
+	db, err := sqlOpen("mysql", strings.TrimSpace(string(dbDSN)))
+	if err != nil {
+		return nil, fmt.Errorf("Could not establish database connection: %s", err)
+	}
+	rows, err := db.Query(`SELECT id,reversedName,notBefore,serial FROM issuedNames where notBefore >= ? and notBefore < ?`, beginTimeStamp, endTimeStamp)
+	if err != nil {
+		return nil, fmt.Errorf("Could not reach database and run query: %s", err)
+	}
+	return rows, nil
+}
+
+func writeTSVData(rows sqlRows, outFile io.Writer) error {
+	for rows.Next() {
+		var (
+			id, rname, notBefore, serial string
+		)
+		if err := rows.Scan(&id, &rname, &notBefore, &serial); err != nil {
+			return err
+		}
+		if _, err := fmt.Fprintf(outFile, "%s\t%s\t%s\t%s\n", id, rname, notBefore, serial); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func compress(outputFileName string) error {
+	gzipCmd := exec.Command("/usr/bin/gzip", outputFileName)
+	err := execRun(gzipCmd)
+	if err != nil {
+		return fmt.Errorf("Could not gzip result file: %s", err)
```

Comment:
Another place where it might be good to include the file name in the error prefix:
```suggestion
		return fmt.Errorf("Could not gzip result file %q: %s", outputFileName, err)
```